File: LeetCode_Algorithm/263(Ugly Number)/263.py
class Solution:
    def isUgly(self, n: int) -> bool:
        # sol1 39ms 83%
        # 한 숫자로 나누어 나누기가 0이 되기 전까지 계속 나눔 
        if n < 1:
            return False
        for k in [2,3,5]:
            while n % k == 0:
                n //= k
        if n == 1:
            return True            
        return False

File: LeetCode_Algorithm/263(Ugly Number)/263.js
var isUgly = function(n) {
  // sol 169ms 97%
  if (n < 1) {
    return false
  }
   for (let i of [2,3,5]) {
      while (n % i == 0) {
        n /= i
      }
   }
   if (i == 1) {
    return true
   } else {
    return false
   }
  
};
